.kokoro/presubmit/downstream-compatibility.sh

```diff
@@ -38,9 +38,14 @@ echo "Install complete. java-shared-dependencies = $SHARED_DEPS_VERSION"
 
 pushd java-shared-dependencies/target
 for repo in ${REPOS_UNDER_TEST//,/ }; do # Split on comma
-  # Perform testing on last release, not HEAD
-  last_release=$(find_last_release_version "$repo")
-  git clone "https://github.com/googleapis/$repo.git" --depth=1 --branch "v$last_release"
+  # Perform testing on HEAD for google-cloud-java
+  if [ $repo == "google-cloud-java" ]; then
+    git clone "https://github.com/googleapis/$repo.git" --depth=1
+  else
+    # Perform testing on last release, not HEAD
+    last_release=$(find_last_release_version "$repo")
+    git clone "https://github.com/googleapis/$repo.git" --depth=1 --branch "v$last_release"
+  fi
```

Comment:
The downstream checks for java-kms were failing with a compilation error due to a removal of a method in grpc-stub:1.58.0.  Needed to make this update to verify the changes in the monorepo: https://github.com/googleapis/google-cloud-java/pull/9854.